subprojects/core/src/test/groovy/org/gradle/api/internal/tasks/DefaultTaskContainerTest.groovy

```diff
@@ -301,7 +301,7 @@ public class DefaultTaskContainerTest extends Specification {
         container.realize()
 
         then:
-        container.size() == 3
+        container.size() == 2
```

Comment:
I cannot think of any use case that would affect an end user. After my change task `b` isn't created anymore. The reason is that we do not walk task dependencies anymore. I believe the following line was only added as a way to verify that we do walk the task dependencies in the `realized()` method.

    aTaskDependency.getDependencies(task) >> { container.create("b"); Collections.singleton(b) }

I cleaned up this test case to reflect the changes and added more integration tests.